### PR TITLE
[WEB-480] Replace linkchecker with htmltest

### PR DESCRIPTION
**Pull request by nsollecito**
### What does this PR do?
Updates Docs preview build to use [htmltest](https://github.com/wjdp/htmltest) in place of BLC

Config is located in `.htmltest.yml`

Sample output can be reviewed here: 
https://gitlab.ddbuild.io/DataDog/documentation/-/jobs/83550127

Currently, only internal links are being checked, internal hash checking is disabled due to the high volume of false-positives from Javascript-centric filters and tags. 

### Motivation
Broken Link Checker is failing to run on Docs. While this is able to be fixed by disabling the gzip option for HTML deploy, running the script on Docs is timing out even after increasing the limit to 3hrs which isn't practical for a continuous build pipeline job. `htmltest` is significantly faster, is actively maintained, and provides more complete coverage. 

### Preview
https://docs-staging.datadoghq.com/nsollecito/linkchecker/

### Additional Notes
- Numerous test options are disabled via config, such as image checking, because there are a number of issues that will need to be address separately. Getting this working for Internal links is the main priority. 
- Some link issues have been fixed
- ~in some cases where a hash is required for filtering by Javascript (such as Integrations filtering by Tag), a `data-proofer-ignore` attribute has been added so that htmltest will not fail on these~

---

### Reviewer checklist
- [x] Review the changed files.
- [x] Review the URLs listed in the [Preview](#preview) section.
- [x] Check images for PII
- [x] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

**Review by sarina-dd — APPROVED** [review]
This is exciting, especially the UI component in the jobs preview! 🥳  Thanks, Nick! LGTM.

**Comment by sarina-dd** [comment]
Hi @nsollecito! Ruth mentioned that some of the tabs listed for the Agent section are actually working as designed. For example:`missing hash --- agent/basic_agent_usage/windows/index.html --> ?tab=gui` does redirect to the gui tab.

Is it possible that we can ignore tabs with the link checker, or find a way to better implement around tabbing so it doesn't populate all of the tabs that are working by design? The previous link checker didn't check tabs to our knowledge.

**Review by bgdeutsch — APPROVED** [review]

**Comment by nsollecito** [comment]
@sarina-dd for now, I've disabled the internal hash check. There were too many false positives. If this is a feature we find useful in the future, we can update the markup to account for URLs that rely on Javascript to function

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -31,10 +31,11 @@ cache: &cache
   key: $CI_COMMIT_REF_SLUG-$CI_PROJECT_DIR
   paths:
     - node_modules/
+    - tmp/.htmltest/
 
 # ================== templates ================== #
 .base_template: &base_template
-  image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/corp-ci:master
+  image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/corp-ci:nsollecito_htmltest
```
Comment (ruthnaebeck) [inline .gitlab-ci.yml:38]: Is this correct for merging to master?

Comment (nsollecito) [reply .gitlab-ci.yml]: @ruthnaebeck I pushed out a new version of the corp-ci build image and pointed this back to master, so we should be good here. Thanks!

**Review by ruthnaebeck — APPROVED** [review]

**Pull request merged**